### PR TITLE
frontend: full screen menu on mobile

### PR DESCRIPTION
**Pull request by shonsirsha**
to improve our app's UI and achieving a more modern look, we'd like to make the sidebar menu / navigation to be of full screen size on mobile.

[Preview] Screen recording on mobile:

https://user-images.githubusercontent.com/28676406/204563461-d42905f8-3f87-4935-9160-481d918f20b6.mov

[Preview] Screenshot on desktop (no change):
Screen Shot 2022-11-29 at 15 45 51

### CODE REVIEW

File: frontends/web/src/components/sidebar/sidebar.css [hunk 1]
```diff
@@ -1,3 +1,15 @@
+.closeButton {
+    background: transparent;
+    border: none;
+    display: none;
+    padding: 12px;
```
Comment (shonsirsha) [inline frontends/web/src/components/sidebar/sidebar.css:5]: padding is not used for spacing, but to increase the button's touchable / surface area which improves UX (user doesnt have to touch exactly the X button, but surrounding is fine).

File: frontends/web/src/components/sidebar/sidebar.css [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+    transform: translateX(14px);
```
Comment (shonsirsha) [inline frontends/web/src/components/sidebar/sidebar.css:6]: this is to "negate" the padding above  (otherwise the X will appear a little bit to the left). 

no big deal, but 14 is used here and not 12 (actual padding val) because of the extra whitespace brought in by the actual `CloseXWhite` component itself.

File: frontends/web/src/components/sidebar/sidebar.css [hunk 3]
```diff
@@ -217,8 +228,14 @@ a.sidebar-active .single img,
...
 @media (max-width: 1199px) {
+    .closeButton {
+        display:block;
+    }
+
     .sidebar {
-        transition: margin-left 0.2s ease;
+        margin-left: -100vw;
+        transition: margin-left 0.3s ease;
+        width: 100vw;
```
Comment (thisconnect) [inline frontends/web/src/components/sidebar/sidebar.css:238]: 1199px is still desktop mode, but with hamburger (unfortunatelly, but one day we fix that).

Could you pick a smaller breakpoint that the navigation menu only goes into fullscreen on mobile portrait, but not on small desktop. Ideally landscape would be as it was before, but mobile portrait would be fullscreen.

**Review by thisconnect — CHANGES_REQUESTED** [review]

File: frontends/web/src/components/sidebar/sidebar.css (hunk 3 above)
Comment (shonsirsha) [reply frontends/web/src/components/sidebar/sidebar.css:238]: Ok. I agree with you.

We currently don't have an exact "mobile" breakpoint, but I'll use a the smallest breakpoint that is currently used in the project which is <=560px.

Comment (shonsirsha) [reply frontends/web/src/components/sidebar/sidebar.css]: Changes made, PTAL 🙏

File: frontends/web/src/components/sidebar/sidebar.tsx [hunk 4]
```diff
@@ -158,13 +159,17 @@ class Sidebar extends Component<Props> {
...
-            <div className="sidebarLogoContainer">
+          <div className="flex flex-between sidebarLogoContainer">
+            <Link
+              to={accounts.length ? '/account-summary' : '/'}
+              onClick={this.handleSidebarItemClick}>
               <AppLogoInverted className="sidebarLogo" />
-            </div>
-          </Link>
+            </Link>
+            <button className="closeButton" onClick={toggleSidebar}>
```
Comment (thisconnect) [inline frontends/web/src/components/sidebar/sidebar.tsx:168]: I just realized this component doesn't use CSS module and the CSS is all global. This component should probably use CSS module and use `className={style.closeButton}`. It would take renaming `sidebar.css` to `sidebar.module.css` and then import style from sidebar.module.css. But can be in a different PR or 2nd commit, up to you.

**Review by thisconnect — CHANGES_REQUESTED** [review]

File: frontends/web/src/components/sidebar/sidebar.tsx (hunk 4 above)
Comment (shonsirsha) [reply frontends/web/src/components/sidebar/sidebar.tsx:168]: Haha, it seems like only this component isn't using css module 😄 https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/frontends/web/src/style/index.css

Will migrate it to be using one here.

File: frontends/web/src/components/sidebar/sidebar.tsx [hunk 5]
```diff
@@ -158,13 +159,17 @@ class Sidebar extends Component<Props> {
       <div className={['sidebarContainer', hidden ? 'forceHide' : ''].join(' ')}>
         <div className={['sidebarOverlay', activeSidebar ? 'active' : ''].join(' ')} onClick={toggleSidebar}></div>
         <nav className={['sidebar', activeSidebar ? 'forceShow' : '', shown ? 'withGuide' : ''].join(' ')}>
-          <Link
-            to={accounts.length ? '/account-summary' : '/'}
-            onClick={this.handleSidebarItemClick}>
-            <div className="sidebarLogoContainer">
+          <div className="flex flex-between sidebarLogoContainer">
```
[the remark this answers is not in the document]
Comment (shonsirsha) [reply frontends/web/src/components/sidebar/sidebar.tsx:162]: sure

File: frontends/web/src/components/sidebar/sidebar.tsx (hunk 4 above)
Comment (thisconnect) [reply frontends/web/src/components/sidebar/sidebar.tsx:168]: wait the styles in `src/style/..` are supposed to be global.
but all css in `src/components/*` are supposed to be CSS modules

**Review by thisconnect — APPROVED** [review]
LGTM

**Comment by shonsirsha** [comment]
Hm, looked like this was merged but the changes aren't on master. 🤔
